BaseLib/uniqueInsert.h

```diff
@@ -15,86 +15,114 @@
 #pragma once
 
 #include <algorithm>
-
+#include <typeinfo>
+#include <typeindex>
 #include "Error.h"
 
-namespace BaseLib {
-
-template<typename Container>
-void uniquePushBack(Container& container, typename Container::value_type const& element)
+namespace BaseLib
+{
+template <typename Container>
+void uniquePushBack(Container& container,
+                    typename Container::value_type const& element)
 {
-    if (std::find(container.begin(), container.end(), element) == container.end())
+    if (std::find(container.begin(), container.end(), element) ==
+        container.end())
         container.push_back(element);
 }
 
+//! Inserts the given \c key with the given \c value into the \c map if an entry
+//! with the
+//! given \c key does not yet exist; otherwise an \c error_message is printed
+//! and the
+//! program is aborted.
+//! Note: The type of \c key must be std::type_index.
+template <typename Map, typename Key, typename Value>
+void insertIfTypeIndexKeyUniqueElseError(Map& map, Key const& key,
```

Comment:
In this file, only this function is  new, and other changes in this file are due to clang-format.